api/slash-z-inner.js

```diff
@@ -0,0 +1,242 @@
+import Prisma from './prisma.js'
+import isPublicSlackChannel from './is-public-slack-channel.js'
+import userIsRestricted from './user-is-restricted.js'
+import channelIsForbidden from './channel-is-forbidden.js'
+import openZoomMeeting from './open-zoom-meeting.js'
+import transcript from './transcript.js'
+import fetch from 'node-fetch'
+
+export default async ({
+  displayName,
+  userId,
+  channelId,
+  responseUrl,
+  skipWarning = false
+}) => {
+  console.log({
+    userId,
+    channelId,
+    restricted: userIsRestricted(userId),
+    forbidden: channelIsForbidden(channelId)
+  })
+
+  if (await userIsRestricted(userId)) {
+    return fetch(responseUrl, {
+      method: 'post',
+      headers: {
+        Authorization: `Bearer ${process.env.SLACK_BOT_USER_OAUTH_ACCESS_TOKEN}`,
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        response_type: 'ephemeral',
+        text: transcript('errors.userIsRestricted')
+      })
+    })
+  }
+
+  if (channelIsForbidden(channelId)) {
```

Comment:
Should this be `await`ed for?

Reply:
It's not async, it just checks if a hardcoded array contains the channel name.

![](https://doggo.ninja/qYY7Ya.png)

Reply:
Forgot that was how it works– good to know!